cql/src/main/java/org/apache/cassandra/stargate/metrics/ClientMetrics.java

```diff
@@ -180,42 +249,95 @@ private int countConnectedClients() {
               (username, count) -> counts.put(username, counts.getOrDefault(username, 0) + count));
     }
 
-    return counts;
+    recordMapToMultiGauge(
+        connectedNativeClientsByUser, counts, username -> Tags.of("username", username));
   }
 
-  private List<Map<String, String>> connectedClients() {
-    List<Map<String, String>> clients = new ArrayList<>();
+  private <T> void recordMapToMultiGauge(
+      MultiGauge gauge, Map<T, ? extends Number> source, Function<T, Tags> tagsFunction) {
+    List<MultiGauge.Row<?>> rows =
+        source.entrySet().stream()
+            .map(entry -> MultiGauge.Row.of(tagsFunction.apply(entry.getKey()), entry.getValue()))
+            .collect(Collectors.toList());
 
-    for (Server server : servers)
-      for (ConnectedClient client : server.getConnectedClients()) clients.add(client.asMap());
+    gauge.register(rows, true);
+  }
 
-    return clients;
+  /**
+   * Resolves a metric name for the micrometer registry.
+   *
+   * @param name metric name
+   * @return full name
+   */
+  private static String metric(String name) {
```

Comment:
I think that we are missing a test case for a backward-compatibility.

Reply:
Adding..

Reply:
Actually it's already tested? All tests in the both unit and int `ClientMetricsTest` confirm the metric names. Or you had something else in mind?

Reply:
yes, the `ClientMetricsTest` tests but I think that only the subset of all metrics (only memory-related)

Reply:
`ClientMetricsTest` unit test every single metric. `ClientMetricsTest` int tests targets few, because there is no need to check all imo, first fill fail anyway. Or?